### PR TITLE
feat: Automated CB based on timestamps, Cron and DB part

### PR DESCRIPTION
**Pull request by ConjunctiveNormalForm**
creates a `Timestamp` Dynamo table that tracks filler hash, lastCheckedPostTimestamp (last time the row is updated), and the timestamp that each filler should be blocked until.

The fade-rate cron job now calculates related timestamps based on recent order data from Redshift and `Timestamp` table

cb

### CODE REVIEW

File: lib/constants.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export const BETA_COMPLIANCE_S3_KEY = 'beta.json';
 export const DYNAMO_TABLE_NAME = {
   FADES: 'Fades',
   SYNTHETIC_SWITCH_TABLE: 'SyntheticSwitchTable',
+  TIMESTAMP: 'Timestamp',
```
Comment (zhongeric) [inline lib/constants.ts:15]: nit but at first glance this table name wasn't very descriptive for me

File: test/crons/fade-rate.test.ts [hunk 2]
```diff
@@ -1,33 +1,103 @@
...
 
 describe('FadeRateCron test', () => {
-  describe('calculateFillerFadeRates', () => {
+  let newFades: FillerFades;
+  beforeAll(() => {
+    newFades = getFillersNewFades(FADES_ROWS, ADDRESS_TO_FILLER, FILLER_TIMESTAMPS, logger);
+  });
+
+  describe('getFillersNewFades', () => {
     it('takes into account multiple filler addresses of the same filler', () => {
```
Comment (zhongeric) [inline test/crons/fade-rate.test.ts:62]: this test name seems outdated?

File: lib/cron/fade-rate.ts [hunk 3]
```diff
@@ -40,49 +51,92 @@ async function main(metrics: MetricsLogger) {
...
     const addressToFillerHash = await webhookProvider.addressToFillerHash();
-    const fillerFadeRate = calculateFillerFadeRates(result, addressToFillerHash, log);
-    log.info({ fadeRates: [...fillerFadeRate.entries()] }, 'filler fade rate');
 
-    const configProvider = new S3CircuitBreakerConfigurationProvider(
-      log,
-      `${FADE_RATE_BUCKET}-prod-1`,
-      FADE_RATE_S3_KEY
+    // aggregated # of fades by filler entity (not address)
+    //  | hash    |  faded  |
```
Comment (zhongeric) [inline lib/cron/fade-rate.ts:69]: nit: fades

**Review by zhongeric — APPROVED** [review]
Looks great!

**Review by rileydcampbell — APPROVED** [review]
nice nice, this lgtm! just for a thought experiment, if we wanted to create a reset button it could just write to the filler row and set the `blockUntilTimestamp` to nothing right?

**Comment by ConjunctiveNormalForm** [comment]
> nice nice, this lgtm! just for a thought experiment, if we wanted to create a reset button it could just write to the filler row and set the `blockUntilTimestamp` to nothing right?

that's right, although I think the main goal of the design is to fully automate things

File: lib/repositories/fades-repository.ts [hunk 4]
```diff
@@ -56,14 +58,15 @@ export class FadesRepository extends BaseRedshiftRepository {
 }
 
 const CREATE_VIEW_SQL = `
-CREATE OR REPLACE VIEW rfqOrdersTimestamp 
+CREATE OR REPLACE VIEW latestRfqs
```
Comment (codyborn) [inline lib/repositories/fades-repository.ts:61]: This gets the latest 20 RFQs for each filler right? The limit of 1k would prevent us from going over 50 fillers. Is this 50 filler limit expected for the foreseeable future?

**Pull request merged**